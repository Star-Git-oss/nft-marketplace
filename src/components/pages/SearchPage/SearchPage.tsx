import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { useTranslation } from 'react-i18next';
import { ApiProxyUri } from '../../../utils/utils';
import NftCard from '../../common/NftCard/NftCard';
import Col from '../../common/ui-library/Col/Col';
import Row from '../../common/ui-library/Row/Row';
import { H4, GradientText } from '../../common/Title/Title';
import useQuery from '../../../hooks/useQuery';

const SearchPage: React.FC<LoadablePageType> = ({ setIsLoading }) => {

  // Nft list in state
  const [nftList, setNftList] = useState([] as NftObjectType[]);

  const query = useQuery();

  useEffect(  () => {
    async function fetchData() {
      setIsLoading(true);
      const res = await axios.get(`${ApiProxyUri}/nfts`);
      setNftList(res.data.nfts);
      setIsLoading(false);
    }
    fetchData();
  }, [setIsLoading]);

  const { t } = useTranslation();

  return (
    <>
      <H4 style={{ margin: '20px' }}>{t('searchPage.resultsFor')} <GradientText> {query.get('q')} </GradientText></H4>
      <Row>
        {nftList?.map((nft, index) => (
          <NftCard key={nft.id} nft={nft} />
        ))}
        <Col size="20" />
      </Row>
    </>
  );
};

export default SearchPage;