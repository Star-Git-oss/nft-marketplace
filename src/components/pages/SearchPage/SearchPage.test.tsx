/* eslint-disable no-console */
import React from 'react';
import axios from 'axios';
import renderer from 'react-test-renderer';
import SearchPage from './SearchPage';

// Known issue: error and warning does not work for console
// https://github.com/facebook/react/issues/7047
const original = console.error;

beforeEach(() => {
  console.error = jest.fn();
});

afterEach(() => {
  console.error = original;
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key) => key }),
}));

const nftList = [
  {
    'id':1,
    'labels': [1, 2, 3],
    'name': 'my NFT 1',
    'quantity': '10 of 10',
    'price': '0.2',
    'image': 'https://picsum.photos/200/300',
    'views': 5201,
    'owner': 'mickael canu',
    'creator': 'inri',
    'collectionName': 'Test name'
  },
  {
    'id':2,
    'labels': [1, 2, 3],
    'name': 'my NFT 2',
    'quantity': '10 of 10',
    'price': '0.5',
    'image': 'https://picsum.photos/201/300',
    'views': 5201,
    'owner': 'mickael canu',
    'creator': 'inri',
    'collectionName': 'Test name'
  },
];

// Axios mock
jest.mock('axios');
const resp = { data: nftList };
axios.get.mockResolvedValue(resp);

// React-router-dom mock
const mockHistoryPush = jest.fn();
const mockSearch = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
  useLocation: () => ({
    search: mockSearch,
  }),
}));

describe('SearchPage snapshots', () => {

  test('SearchPage renders correctly', () => {

    const tree = renderer
      .create(
        <SearchPage setIsLoading={()=> jest.fn()} />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

});