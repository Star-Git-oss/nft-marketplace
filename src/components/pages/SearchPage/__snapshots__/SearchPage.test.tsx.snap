// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage snapshots SearchPage renders correctly 1`] = `
Array [
  <h4
    className="sc-jrAGrp ftfZuI"
    style={
      Object {
        "margin": "20px",
      }
    }
  >
    searchPage.resultsFor
     
    <span
      className="sc-dQppl jezuye"
    >
       
       
    </span>
  </h4>,
  <div
    className="sc-gKsewC krDCXJ"
  >
    <div
      className="sc-jSgupP gUEbOS"
      size="20"
    />
  </div>,
]
`;
